bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -1059,7 +1059,7 @@ void ReplicaImp::onMessage<PrePrepareMsg>(PrePrepareMsgUPtr msg) {
              "etc...)");
     return;
   }
-  PrePrepareMsgShPtr message(msg.release());
+  PrePrepareMsgShPtr message(std::move(msg));
```

Comment:
A more efficient way to do this, is to bind straight into a shared pointer and remove this line:
```
template <>
void ReplicaImp::onMessage<PrePrepareMsg>(PrePrepareMsgShPtr msg) {
```

Reply:
That would change the template method onMessage signature